### PR TITLE
Quit if follow fails because of not configured addon

### PR DESCRIPTION
**Pull request by annando**
When an addon fails we now quit the following.

### CODE REVIEW

**Comment by MrPetovan** [comment]
Fixes #4362

File: src/Model/Contact.php [hunk 1]
```diff
@@ -1151,6 +1151,11 @@ public static function createFromProbe($uid, $url, $interactive = false, $networ
 
 		Addon::callHooks('follow', $arr);
 
+		if (empty($arr)) {
+			$result['message'] = L10n::t('Contact cannot be added.');
```
Comment (MrPetovan) [inline src/Model/Contact.php]: Can you please be more specific about the error? Ask to check the credentials if it's an addon-provided network for example.

Comment (annando) [reply src/Model/Contact.php]: Then the message had to be generated by the addon.

Comment (MrPetovan) [reply src/Model/Contact.php]: It is a solution, but this check is after the Addon hook, so you can just write something general about addon-provided networks and be done with it.

Comment (annando) [reply src/Model/Contact.php]: I don't have a clue for a nice message.

Comment (zeroadam) [reply src/Model/Contact.php]: The contact could not be added. It is possible you are attempting to add a contact that requires an Addon that is not enabled on this system.

Comment (annando) [reply src/Model/Contact.php]: Done

File: src/Model/Contact.php [hunk 2]
```diff
@@ -1152,7 +1152,7 @@ public static function createFromProbe($uid, $url, $interactive = false, $networ
 		Addon::callHooks('follow', $arr);
 
 		if (empty($arr)) {
-			$result['message'] = L10n::t('Contact cannot be added.');
+			$result['message'] = L10n::t('The contact could not be added. It is possible you are attempting to add a contact that requires an Addon that is not enabled on this system.');
```
Comment (MrPetovan) [inline src/Model/Contact.php]: The addon was enabled, but the user didn't configure it. If it wasn't enabled the hook wouldn't be called at all.

Comment (annando) [reply src/Model/Contact.php]: Changed again :-)

File: src/Model/Contact.php [hunk 3]
```diff
@@ -1152,7 +1152,7 @@ public static function createFromProbe($uid, $url, $interactive = false, $networ
 		Addon::callHooks('follow', $arr);
 
 		if (empty($arr)) {
-			$result['message'] = L10n::t('The contact could not be added. It is possible you are attempting to add a contact that requires an Addon that is not enabled on this system.');
+			$result['message'] = L10n::t('The contact could not be added. It is possible you are attempting to add a contact that requires an Addon that is not configured on this system.');
```
Comment (MrPetovan) [inline src/Model/Contact.php]: "The contact could not be added. Please check the relevant network credentials in your Settings -> Social Networks page."

Comment (annando) [reply src/Model/Contact.php]: But possibly the addon wasn't configured by the administrator.

Comment (MrPetovan) [reply src/Model/Contact.php]: Indeed, but it's less likely that the admin enabled the addon and didn't configure it than a random user just tried to add a twitter user before signing in with Twitter.

**Pull request merged**